Upstream-Status: Submitted [https://gitlab.gnome.org/GNOME/glib/-/merge_requests/2177]
Signed-off-by: Ross Burton <ross.burton@arm.com>

From 289f8be1b397a453cfcf35641455f3ae5fb4faeb Mon Sep 17 00:00:00 2001
From: Ross Burton <ross.burton@arm.com>
Date: Tue, 6 Jul 2021 19:26:03 +0100
Subject: [PATCH] gio/tests/g-file-info: don't assume million-in-one events
 don't happen

The modification time test creates a file, gets the modification time in
seconds, then gets the modification time in microseconds and assumes
that the difference between the two has to be above 0.

As rare as this may be, it can happen:

$ stat g-file-info-test-50A450 -c %y
2021-07-06 18:24:56.000000767 +0100

Change the test to simply assert that the difference not negative to
handle this case.
---
 gio/tests/g-file-info.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gio/tests/g-file-info.c b/gio/tests/g-file-info.c
index c11c50462..fd0c64b55 100644
--- a/gio/tests/g-file-info.c
+++ b/gio/tests/g-file-info.c
@@ -178,7 +178,7 @@ test_g_file_info_modification_time (void)
   g_assert_nonnull (dt_usecs);
 
   ts = g_date_time_difference (dt_usecs, dt);
-  g_assert_cmpint (ts, >, 0);
+  g_assert_cmpint (ts, >=, 0);
   g_assert_cmpint (ts, <, G_USEC_PER_SEC);
 
   /* Try round-tripping the modification time. */
-- 
2.25.1

